robottelo/host_helpers/capsule_mixins.py

```diff
@@ -67,20 +72,51 @@ def wait_for_sync(self, timeout=600, start_time=None):
         # is started (or finished already)
         if start_time is None:
             start_time = datetime.utcnow().replace(microsecond=0)
+        # 1s margin of safety for rounding
+        start_time = (start_time - timedelta(seconds=1)).strftime('%Y-%m-%d %H:%M:%S UTC')
         logger.info(f"Waiting for capsule {self.hostname} sync to finish ...")
         sync_status = self.nailgun_capsule.content_get_sync()
         logger.info(f"Active tasks {sync_status['active_sync_tasks']}")
-        assert (
```

Comment:
What was wrong with this assertion? You changed it to something more complicated at lines 82 to 98 and I can't see a benefit.

Reply:
I think there is also some timing issue, the `start_time` should be before the promote task and passed from the test, as the bulkAction sync task seems to be started before it finishes? 
Not 100% sure about what's going on yet, just put up some new changes that pass locally.